### PR TITLE
Changes necessary to build with clang-omp on Mac 10.10.5

### PR DESCRIPTION
**Pull request by buske**
Fixes the following build issues encountered:

Undeclared __gnu_parallel:

```
In file included from algorithms.cpp:27:
In file included from ./algorithms.h:28:
In file included from ./gcsa.h:28:
In file included from ./files.h:28:
In file included from ./support.h:28:
./utils.h:221:32: error: use of undeclared identifier '__gnu_parallel'
  std::sort(first, last, comp, __gnu_parallel::balanced_quicksort_tag());
                               ^
./utils.h:235:26: error: use of undeclared identifier '__gnu_parallel'
  std::sort(first, last, __gnu_parallel::balanced_quicksort_tag());
                         ^
./utils.h:247:32: error: use of undeclared identifier '__gnu_parallel'
  std::sort(first, last, comp, __gnu_parallel::multiway_mergesort_tag());
                               ^
./utils.h:258:26: error: use of undeclared identifier '__gnu_parallel'
  std::sort(first, last, __gnu_parallel::multiway_mergesort_tag());
                         ^
./utils.h:269:32: error: use of undeclared identifier '__gnu_parallel'
  std::sort(first, last, comp, __gnu_parallel::sequential_tag());
                               ^
./utils.h:280:26: error: use of undeclared identifier '__gnu_parallel'
  std::sort(first, last, __gnu_parallel::sequential_tag());
                         ^
6 errors generated.
make[1]: *** [algorithms.o] Error 1
make: *** [lib/libgcsa2.a] Error 2
```

Undefined symbols because of direct use of static const (when passing by const reference):

```
Undefined symbols for architecture x86_64:
  "__ZN4gcsa10InputGraph7UNKNOWNE", referenced from:
      __ZN4gcsa9parseKMerERKNSt3__112basic_stringIcNS0_11char_traitsIcEENS0_9allocatorIcEEEE in libgcsa2.a(files.o)
  "__ZN4gcsa15ReverseTrieNode11SEED_LENGTHE", referenced from:
      __ZN4gcsa10countKMersERKNS_4GCSAEyb in libgcsa2.a(algorithms.o)
  "__ZN4gcsa3Key10MAX_LENGTHE", referenced from:
      __ZN4gcsa3Key6decodeEyyRKNS_8AlphabetE in libgcsa2.a(support.o)
  "__ZN4gcsa9PathLabel12LABEL_LENGTHE", referenced from:
      __ZN4gcsa17MergedGraphReader9intersectERKNS_9PathLabelES3_y in libgcsa2.a(gcsa.o)
ld: symbol(s) not found for architecture x86_64
```

http://stackoverflow.com/questions/5391973/undefined-reference-to-static-const-int
http://stackoverflow.com/questions/2605520/c-where-to-initialize-static-const

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -8,7 +8,7 @@ SDSL_DIR=../sdsl-lite
 #VERIFY_FLAGS=-DVERIFY_CONSTRUCTION
 
 # Multithreading with OpenMP and libstdc++ Parallel Mode.
-PARALLEL_FLAGS=-fopenmp -pthread -D_GLIBCXX_PARALLEL
+PARALLEL_FLAGS=-fopenmp -pthread
```
Comment (buske) [inline Makefile:11]: There's probably a better way of doing this so that the flag is only enabled when using gcc.

Comment (jltsiren) [inline Makefile:11]: `_GLIBCXX_PARALLEL` must be defined in a compiler flag (or at least before including any STL headers).

My long-term plan is to get rid of libstdc++ Parallel Mode (and OpenMP) completely. I just need a multithreaded in-place replacement for `std::sort` using C++ threads. After that, any compiler that can compile sdsl-lite should also be able to compile gcsa2.

Comment (buske) [inline Makefile:11]: Is it possible to remove the flag for clang builds, until gcsa2 can migrate from libstdc++?

Comment (jltsiren) [inline Makefile:11]: Removing the flag makes index construction slower, but everything should still work correctly.

**Comment by buske** [comment]
The Makefile change should now only affect clang users. Anything else this needs before merge? Necessary for downstream clang support in `vg`.

**Pull request merged**